From 0d0c44ee804fbdf3fca168a4e015295f46df6788 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Wed, 15 Dec 2021 04:46:21 +0200
Subject: [PATCH] Revert "[SW-63186] gaudi2: security: set axprot for hmmu"

Moving axprot configuration of hmmu to zephyr post isolation.
This is required as the engines which have isolation logic must
be accessed/configured post isolation configuration.

This reverts commit 935691d0f018609203d40e6e207b56cbe8e8e9c0.

Change-Id: Ie277c3f79e83d8a9bc47ca08d68bed000664b879
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 4586686..143c913 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -35,7 +35,6 @@
 #include <dcore0_vdec0_mstr_if_core_lbw_regs.h>
 #include <rot0_mstr_if_core_hbw_regs.h>
 #include <rot0_mstr_if_core_lbw_regs.h>
-#include <dcore0_hmmu0_mstr_if_core_hbw_regs.h>
 #include <pmmu_hbw_mstr_if_core_hbw_regs.h>
 #include <arc_farm_kdma_mstr_if_core_hbw_regs.h>
 #include <arc_farm_kdma_mstr_if_core_lbw_regs.h>
@@ -207,7 +206,6 @@
 enum preboot_sec_axprot_regs {
 	AXPROT_VDEC,
 	AXPROT_ROT,
-	AXPROT_HMMU,
 	AXPROT_MME,
 	AXPROT_PDMA,
 };
@@ -925,10 +923,6 @@
 #define ROT_RTR_INST_NUM	1
 #define ROT_INST_NUM		2
 #define ROT_INST_DIST		0x10000
-#define HMMU_RTR_INST_NUM	NUM_OF_DCORES
-#define HMMU_RTR_INST_DIST	0x10000
-#define HMMU_INST_NUM		NUM_OF_DCORES
-#define HMMU_INST_DIST		DCORE_DIST
 #define MME_SBTE_RTR_INST_NUM	5
 #define MME_SBTE_RTR_INST_DIST	0x8000
 #define MME_WB_RTR_INST_NUM	2
@@ -957,14 +951,6 @@
 		.dcore_num = ROT_INST_NUM,
 		.inst_dist = 0,
 		.inst_num = ROT_RTR_INST_NUM,
-	},
-	[AXPROT_HMMU] = {
-		.base_reg = mmDCORE0_HMMU0_MSTR_IF_CORE_HBW_AXPROT_OVRD,
-		.value = AXPROT_SEC_VAL,
-		.dcore_dist = HMMU_INST_DIST,
-		.dcore_num = HMMU_INST_NUM,
-		.inst_dist = HMMU_RTR_INST_DIST,
-		.inst_num = HMMU_RTR_INST_NUM,
 	},
 	[AXPROT_MME] = {
 		.base_reg = mmDCORE0_MME_SBTE0_MSTR_IF_CORE_HBW_AXPROT_OVRD,
@@ -1019,13 +1005,6 @@
 
 	/* LBW AXPROT */
 	preboot_axprot_regs_init(AXPROT_ROT);
-}
-
-static void preboot_secure_hmmu(void)
-{
-	/* Configure hmmu to secure level */
-	/* HBW AXPROT */
-	preboot_axprot_regs_init(AXPROT_HMMU);
 }
 
 static void preboot_secure_pmmu(void)
@@ -1098,7 +1077,6 @@
 		hl_dev_prot_bits_init();
 		preboot_secure_vdec();
 		preboot_secure_rot();
-		preboot_secure_hmmu();
 		preboot_secure_pmmu();
 		preboot_secure_arc_farm();
 		preboot_secure_trace();
